include/picongpu/particles/startPosition/OnePositionImpl.hpp

```diff
@@ -122,6 +122,8 @@ namespace detail
                     m_weighting
                 );
 
+            result = realParticlesPerCell > 0.0_X ? result : 0.0_X;
```

Comment:
OK it takes a while until I got why you add this line.
In the case where we have no real particles you must set the number of particles to zero.

Since this is a special case if `hasWeighting == false`, please add this line into the else branch if the if condition.

Reply:
IMO #2831 should fully solve the issue and I would say this PR is not needed. @ax3l  I am wrong?

Reply:
I checked it again and #2831 [this line](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2831/files#diff-e80f8636fbb31c1a3c7bc8a2f1366234R246) avoids that `numberOfMacroParticles` is called for any density `<= 0`

Reply:
Yes, as I wrote I expect to #2831 have fixed this already.
@x113f reports the issue on `0.4.2` which doesn't have that fix already.